apps/jobs/views.py

```diff
@@ -745,3 +749,354 @@ def update_submission(request, challenge_pk):
+            logger.exception('Cannot get queue: {}'.format(queue_name))
+
+    try:
+        message = queue.Message(receipt_handle)
+        message.delete()
+        response_data = {
+            'success': 'The message is successfully deleted from the queue!'
+        }
+        return Response(response_data, status=status.HTTP_200_OK)
+    except botocore.exceptions.ClientError as ex:
+        response_data = ex
+        logger.exception("The SQS message is not deleted due to {}".format(ex))
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['PUT'])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def update_submission_status_by_queue_name(request, challenge_pk, submission_pk, queue_name):
+    """
+    API to update the submission status by the challenge hosts
+
+    Arguments:
+        challenge_pk  -- The challenge ID whose submision is to be updated
+        submission_pk  -- The submission ID which is to be updated
+        queue_name  -- The unique authentication token provided by challenge hosts
+
+    """
+
+    try:
+        submission = Submission.objects.get(pk=submission_pk)
+    except Submission.DoesNotExist:
+        response_data = {
+            'error': 'Submission {} does not exists!'.format(submission_pk)
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if not is_user_a_host_of_challenge(request.user, challenge_pk):
+        response_data = {
+            'error': 'Sorry, you are not authorized to access this resource!'
+        }
+        return Response(response_data, status=status.HTTP_401_UNAUTHORIZED)
+
+    challenge_queue_name = submission.challenge_phase.challenge.queue
+
+    if (challenge_queue_name == queue_name):
+        serializer = SubmissionSerializer(submission,
+                                          data=request.data,
+                                          context={'request': request},
+                                          partial=True)
+        if serializer.is_valid():
+            serializer.save()
+            response_data = serializer.data
+            return Response(response_data, status=status.HTTP_200_OK)
+        else:
+            response_data = serializer.errors
+            return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+    else:
+        response_data = {
+            'error': 'Sorry, you are not authorized to update this submission!'
+        }
+        return Response(response_data, status=status.HTTP_401_UNAUTHORIZED)
+
+
+@api_view(['GET'])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_submission_by_pk_using_queue_name(request, submission_pk, queue_name):
```

Comment:
We dont need this API. Can you please use the API that takes the submission id and gives the submission API?